### DISCUSSION TITLE
Error 25 at random times

### DISCUSSION THREAD

**Discussion by enriquemilio** [fra589/grbl-Mega-5X#267] [Q&A]
Hello,

First thanks for this great code

I have a problem that triggers error 25 at random times, using the same gcode file over, it fails at different times, it is never in the same line of code. 

My setup is 1 stepper for Z, one for X and 2 for Y axis. The Y axis is cloned to the 4th axis, I am able to home both Y's independently so they auto square.

Simple files run just fine, but gcode files with adaptive strategies from Fusion 360 is when the error is generated

Anyone have had the same problem?
How can I fix this? 

Thanks in advance for any help!

**Comment by enriquemilio** [comment]
Algo I forgot to mention that I am not using the RAMPS board, I have external stepper drivers

**Comment by fra589** [comment]
Hi @enriquemilio,  
  
Random errors generally occur owing to parameters flash memory corruption.  
Try to reset to factory default parameters with the Grbl $RST=* command, then resend your specifics settings.  
If that doesn't fix your problems, please post the GCode file that generates the errors...  
  
@++;  
Gauthier.

**Reply by enriquemilio** [reply]
Hello Gauthier, 

I did try that before and it didn't work. 

I read somewhere that it may be related to the clock speed of the mega, so I tried lowering the feeds ands speed and suddenly the error went away, tried again with faster speed and the error came back.

I haven't been able to find a solution for this particular case. What I did is to use the original grbl for the mega board and it works prefect with even faster speed that what I machine at. But I don't like that solution as it requires to bridge the to Y motor, hence the auto square when homing is lost. 

Also I had modified the pins for the step, direction, enable and limits for all axis. I don't know if that may affect this issue.

Do you have a suggestion for me to try?

**Comment by fra589** [comment]
Hi @enriquemilio,  
  
Please, post your configuration (config.h & cpu_map.h files, $I and $$ command result), so I will can study your Grbl's behaviors ans try to understand...  
Also post the problematic GCode file so that the question will be completely in my brain.  
  
@++;  
Gauthier.

**Comment by enriquemilio** [comment]
Hello Gauthier,

Very nice to hear from you, thanks for all your time and help.

Currently I don't have the $I information has I am temporarily using other build, if you need it, i can flash it again and send it to you.

[cpu_map.h.txt](https://github.com/fra589/grbl-Mega-5X/files/8563796/cpu_map.h.txt)
[config.h.txt](https://github.com/fra589/grbl-Mega-5X/files/8563797/config.h.txt)


$0 = 10    (Step pulse time, microseconds)
$1 = 254    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 11    (Step direction invert, mask)
$4 = 1    (Invert step enable pin, boolean)
$5 = 1    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 0.020    (Junction deviation, millimeters)
$12 = 0.015    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 1    (Homing cycle enable, boolean)
$23 = 11    (Homing direction invert, mask)
$24 = 100.000    (Homing locate feed rate, mm/min)
$25 = 1000.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 5.000    (Homing switch pull-off distance, millimeters)
$30 = 24000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$100 = 200.000    (X-axis travel resolution, step/mm)
$101 = 200.000    (Y-axis travel resolution, step/mm)
$102 = 200.000    (Z-axis travel resolution, step/mm)
$103 = 200.000   
$110 = 2750.000    (X-axis maximum rate, mm/min)
$111 = 2750.000    (Y-axis maximum rate, mm/min)
$112 = 750.000    (Z-axis maximum rate, mm/min)
$113 = 2750.000   
$120 = 50.000    (X-axis acceleration, mm/sec^2)
$121 = 50.000    (Y-axis acceleration, mm/sec^2)
$122 = 50.000    (Z-axis acceleration, mm/sec^2)
$123 = 50.000   
$130 = 746.000    (X-axis maximum travel, millimeters)
$131 = 1100.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)
$133 = 1100.000   
ok

**Comment by enriquemilio** [comment]
And here is the GCode i was using.

[PROP.TOP.FULL.nc.txt](https://github.com/fra589/grbl-Mega-5X/files/8563850/PROP.TOP.FULL.nc.txt)

**Comment by fra589** [comment]
Hi @enriquemilio,  
  
I tested several times your GCode file on my develop Arduino Mega card with same $$ parameters and axes parameters than you without any error!  
  
The first idea which come to me is a communication error... 
Perhaps your communication line is too long? Try to lower BAUD_RATE it in config.h to see if it can be better...

I dont think your problem can come from your cpu_map.h changes. Those changes affect only the destination variable of electronics signals, they does not affect the cpu load.
As you think, may be your problem come with a too high speed... 
But as I see, the config.h file you posted is inconsistent with the $$ output you also posted..
The $I output may confirm this, but it seems that the $$ output is a 4 axes configuration while you have `#define N_AXIS 5` in the config.h file...
So, for your need (3 axis X, Y & Z with the 4th axis as Y clone), you should have in config.h :
```
#define N_AXIS 4        // Number of axes (3 to 6)
#define N_AXIS_LINEAR 4 // Number of linears axis, must be <= N_AXIS

```
Another thing which affect the max speed ability is that you enabled the #define ENABLE_RAMPS_HW_LIMITS in cpu_map.h.
Normally this is useless if you have correctly set the maximum travels parameters for all axis, and if your machine is correctly homed.
This brings more trouble than it protects against travels overruns.
Since ENABLE_RAMPS_HW_LIMITS is not available in the original grbl-Mega, try to disable it to see if it work better...

@++;  
Gauthier
